lib/web_ui/lib/src/engine/surface/surface.dart

```diff
@@ -483,6 +483,39 @@ enum PersistedSurfaceState {
   released,
 }
 
+class PersistedSurfaceException implements Exception {
+  PersistedSurfaceException(this.surface, this.message);
+
+  final PersistedSurface surface;
+  final String message;
+
+  @override
+  String toString() {
+    if (assertionsEnabled) {
+      return '${surface.runtimeType}: $message';
+    }
+    return super.toString();
+  }
+}
+
+/// Verifies that the [surface] is in one of the valid states.
+///
+/// This function should be used inside an assertion expression.
+bool debugAssertSurfaceState(PersistedSurface surface, PersistedSurfaceState state1, [PersistedSurfaceState state2, PersistedSurfaceState state3]) {
+  final List<PersistedSurfaceState> validStates = [ state1, state2, state3 ];
+
+  if (validStates.contains(surface.state)) {
```

Comment:
Is `surface.state` nullable? If it's null, it could mistakenly pass this check.

Reply:
It is not nullable. We have an assert for that.

Reply:
I can't wait for Null Safety!